content/examples/recursion.md

```diff
@@ -0,0 +1,23 @@
+---
+title: "recursion"
+date: 2023-06-21T14:00:01+0000
+draft: false
+---
+
+Cairo uses recursion as the principal method of iteration.
+
+Here is how we can calculate the factorial using this method.
+
+```rust {.codebox}
+
+fn main() -> felt252 {
```

Comment:
main expects a felt252 but nothing is returned.
Maybe try adding a test that calls factorial and check the result with `assert`